AK/Math.h

```diff
@@ -220,6 +220,25 @@ constexpr T rint(T x)
 #    endif
 #elif ARCH(AARCH64)
     AARCH64_INSTRUCTION(frintx, x);
+#elif ARCH(RISCV64)
+    if constexpr (IsSame<T, float>) {
+        i64 r;
+        // FIXME: This saturates at 64-bit integer boundaries; see Table 11.4 (RISC-V Unprivileged ISA V20191213)
```

Comment:
You could do a range check for that, as anything close to the integer limits is guaranteed to be an integer

Reply:
I think this should work now